src/main/java/org/jenkinsci/plugins/workflow/steps/SynchronousResumeNotSupportedException.java

```diff
@@ -33,8 +33,18 @@
  */
 public class SynchronousResumeNotSupportedException extends Exception {
 
+    private static final String message = "Resume after a restart not supported for non-blocking synchronous steps";
+
     public SynchronousResumeNotSupportedException() {
-        super("Resume after a restart not supported for non-blocking synchronous steps");
+        super(message);
+    }
+
+    public SynchronousResumeNotSupportedException(String stepDisplayFunctionName) {
+        super(
+            stepDisplayFunctionName == null
+            ? message
+            : message + ", failed step is " + stepDisplayFunctionName + ", you may wrap this block into a retry step "
+              + "with nonresumable condition");
```

Comment:
I think something like 
> Step $name is a non-blocking synchronous step, it doesn't support resume after a restart. You may wrap it within retry(conditions: [nonresumable()], count: 2) {...}

would be more straight to the point.

Reply:
(Can also mention that in Declarative syntax, use the `retries` option to an `agent` directive.)

Reply:
makes sense, now added both syntax message. ty